reference/security/encryption-at-rest.md

```diff
@@ -0,0 +1,126 @@
+2. Data key. The data key is generated by TiKV and is the key actually being used to encrypt data. The data key is automatically rotated by TiKV.
+
+The same master key can be shared by multiple instances of TiKV. The recommended way to provide a master key in production is via AWS KMS. User would need to create a CMK through AWS KMS, and then provide the CMK key id to TiKV in config file. The TiKV process would need access to the KMS CMK while it is running, which can be done by using [IAM](https://aws.amazon.com/iam/). If TiKV fail to get access to the KMS CMK, it will failed to start or restart. If TiKV loss access to the KMS CMK while it is running, data key rotation will be temporarily disabled. Please refer to AWS documentation for KMS and IAM usage.
+
+Alternatively, if using custom key is desired, supplying master key via file is also supported. The file need to contain a 256 bits (or 32 bytes) key encoded as hex string. The file should end with a newline (i.e. "\n") and contain nothing else. Persistenting the key on disk, however, leaks the key, so the key file is only suitable to be stored on tempfs.
+
+Data keys are generated by TiKV and pass to underlying storage engine (i.e. RocksDB). All files written by RocksDB, including SST files, WAL files, and the MANIFEST file,  is encrypted by the current data key. Other temporary files used by TiKV that may include user data are also encrypted using the same data key. By default, data keys are automatically rotated by TiKV every week, for which the period is configurable. On key rotation, TiKV does not rewrite all existing files to replace the key, but RocksDB compaction are expected to rewrite old data into new data files, with the most recent data key, if the cluster gets constant write workload. TiKV keeps track of the key and encryption method used to encrypt each of of files and use the information to decrypt the content on reads.
+
+Regardless of data encryption method, data keys are encrypted using AES256 in GCM mode for additional authentication. This is why when passing master key from file the key has to be of 256 bits.
+
+### Configuring Encryption
+
+To enable encryption, you can add the encryption section in TiKV's config file:
+
+```
+[security.encryption]
+data-encryption-method = aes128-ctr
+data-key-rotation-period = 7d
+```
+
+Possible values for `data-encryption-method` are "aes128-ctr", "aes-192-ctr", "aes256-ctr" and "plaintext". The default value is "plaintext", which means encryption is not turned on. `data-key-rotation-period` defines how often TiKV rotate data key. Encryption can be turned on for a fresh TiKV cluster, or an existing TiKV cluster, though only data written after encryption being enabled are guaranteed to get encrypted. To disable encryption, remove `data-encryption-method` in config file, or reset it to "plaintext", and restart TiKV. To change encryption method, update `data-encryption-method` in config file and restart TiKV.
```

Comment:
```suggestion
Possible values for `data-encryption-method` are "aes128-ctr", "aes-192-ctr", "aes256-ctr" and "plaintext". The default value is "plaintext", which means encryption is not turned on. `data-key-rotation-period` defines how often TiKV rotates the data key. Encryption can be turned on for a fresh TiKV cluster, or an existing TiKV cluster, though only data written after encryption is enabled is guaranteed to be encrypted. To disable encryption, remove `data-encryption-method` in the config file, or reset it to "plaintext", and restart TiKV. To change encryption method, update `data-encryption-method` in config file and restart TiKV.
```

Reply:
If I change it back to plaintext, at what point will all data be decrypted?

Reply:
along with rocksdb compaction, which is done lazily.